PencilCase/PencilCase/scripts/Models/Order.ts

```diff
@@ -22,17 +22,17 @@ export class Order {
         this.type = ko.observable(type);
         switch (type) {
             case OrderTypes.Retail:
-                this.price = ko.observable(product.RetailPrice);
+                this.price = ko.observable(price == undefined ? product.RetailPrice : price);
```

Comment:
For Retail page, use the specific price value, if not defined, use retail price as default.